iceoryx_platform/linux/include/iceoryx_platform/platform_settings.hpp

```diff
@@ -41,8 +41,13 @@ constexpr const char IOX_LOCK_FILE_PATH_PREFIX[] = "/tmp/";
 constexpr uint64_t MAX_USER_NAME_LENGTH = 32;
 constexpr uint64_t MAX_GROUP_NAME_LENGTH = 32;
 
+#if __cplusplus >= 201703L
+template <typename C, typename... Cargs>
+using invoke_result = std::invoke_result<C(Cargs...)>;
```

Comment:
Oh, I think this must be
```suggestion
using invoke_result = std::invoke_result<C, Cargs...>;
```

See also original macOS define.

Reply:
Changed for all the platforms